src/linker/Linker.Steps/BodySubstitutionParser.cs

```diff
@@ -0,0 +1,201 @@
+using System;
+using System.Diagnostics;
+using System.Linq;
+using System.Xml.XPath;
+using Mono.Cecil;
+
+namespace Mono.Linker.Steps
+{
+	public class BodySubstitutionParser : ProcessLinkerXmlStepBase
+	{
+		SubstitutionInfo _substitutionInfo;
+
+		public BodySubstitutionParser (XPathDocument document, string xmlDocumentLocation)
+			: base (document, xmlDocumentLocation)
+		{
+		}
+
+		public BodySubstitutionParser (XPathDocument document, EmbeddedResource resource, AssemblyDefinition resourceAssembly, string xmlDocumentLocation = "")
+			: base (document, resource, resourceAssembly, xmlDocumentLocation)
+		{
+		}
+
+		public SubstitutionInfo Parse (LinkContext context)
+		{
+			_substitutionInfo = new SubstitutionInfo ();
+			Process (context);
+			return _substitutionInfo;
+		}
+
+		public void Parse (LinkContext context, SubstitutionInfo xmlInfo)
```

Comment:
If we need the "in -> modify -> out " overload (this one), then I would only have that one. The caller can trivially create new info as input. It's better than to have to reason about which overload do I need to call.

Reply:
Good idea, fixed!